conan/tools/meson/toolchain.py

```diff
@@ -158,14 +158,13 @@ def _add_apple_flags(self):
         if not sdk_path and self.cross_build:
             raise ConanException("You must provide a valid SDK path for cross-compilation.")
 
-        # TODO: Delete this os_sdk check whenever the _guess_apple_sdk_name() function disappears
-        os_sdk = conanfile.settings.get_safe('os.sdk')
-        if not os_sdk and os_ != "Macos":
-            raise ConanException("Please, specify a suitable value for os.sdk.")
-
+        os_sdk = conanfile.settings.get_safe("os.sdk", "macosx")
```

Comment:
I don't think this is fully correct. The ``os.sdk`` must be defined for ``iOS``, ``watchOS``, etc, so it is not that we want to fall back to ``macosx`` if the ``os.sdk`` is not defined, that only happens for ``Macos``.